### PR TITLE
fix(tsc): fix record body bug in tracing

### PR DESCRIPTION
**Pull request by Qinyouzeng**
- Read the body stream instead of synchronization.
 - if body size is less or equal 200k, tracing will record body ,else record in log with base64 encoding.

### CODE REVIEW

File: src/Contrib/StackSdks/Masa.Contrib.StackSdks.Tsc/Tracing/OpenTelemetryInstrumentationOptions.cs [hunk 1]
```diff
@@ -51,4 +53,38 @@ public class OpenTelemetryInstrumentationOptions
     /// Build trace callback, allow to supplement the build process
     /// </summary>
     public Action<TracerProviderBuilder> BuildTraceCallback { get; set; }
+
+    public void SetMaxBodySize(string val)
```
Comment (zhenlei520) [inline src/Contrib/StackSdks/Masa.Contrib.StackSdks.Tsc/Tracing/OpenTelemetryInstrumentationOptions.cs:57]: meaningless parameter name val

File: src/Contrib/StackSdks/Masa.Contrib.StackSdks.Tsc/Tracing/ActivityExtensions.cs [hunk 2]
```diff
@@ -16,26 +16,26 @@ public static async Task<Activity> AddMasaSupplement(this Activity activity, Htt
...
         activity.SetTag(OpenTelemetryAttributeName.Http.SCHEME, httpRequest.RequestUri?.Scheme);
         activity.SetTag(OpenTelemetryAttributeName.Host.NAME, Dns.GetHostName());
 
         if (httpRequest.Content is not null)
         {
-            var st = await httpRequest.Content.ReadAsStreamAsync();
-            activity.SetTag(OpenTelemetryAttributeName.Http.REQUEST_CONTENT_BODY, await st.ReadAsStringAsync(GetHttpRequestMessageEncoding(httpRequest)));
+            var streamTask = httpRequest.Content.ReadAsStreamAsync();
+            streamTask.Wait();
+            SetActivityBody(activity, streamTask.Result, GetHttpRequestMessageEncoding(httpRequest)).Wait();
```
Comment (zhenlei520) [inline src/Contrib/StackSdks/Masa.Contrib.StackSdks.Tsc/Tracing/ActivityExtensions.cs:34]: Synchronous call to asynchronous? Use wait directly?

**Review by zhenlei520 — CHANGES_REQUESTED** [review]

**Comment by zhenlei520** [comment]
[6 Code Smells](https://sonarcloud.io/project/issues?id=masastack_MASA.Framework&pullRequest=538&resolved=false&types=CODE_SMELL)

File: src/Contrib/StackSdks/Masa.Contrib.StackSdks.Tsc/Tracing/ActivityExtensions.cs [hunk 3]
```diff
@@ -93,10 +93,10 @@ private static async Task SetActivityBody(Activity activity, Stream inputSteam,
         (long length, string? body) = await inputSteam.ReadAsStringAsync(encoding);
 
         if (length <= 0)
-            return;        
+            return;
         if (length - OpenTelemetryInstrumentationOptions.MaxBodySize > 0)
         {
-            OpenTelemetryInstrumentationOptions.Logger?.LogInformation(body);
+            OpenTelemetryInstrumentationOptions.Logger?.LogInformation("Request body in base64 encode:{Body}", body);
```
Comment (zhenlei520) [inline src/Contrib/StackSdks/Masa.Contrib.StackSdks.Tsc/Tracing/ActivityExtensions.cs:97]: `Request body in base64 encode: {Body}`

File: src/Contrib/StackSdks/Masa.Contrib.StackSdks.Tsc/Tracing/OpenTelemetryInstrumentationOptions.cs [hunk 4]
```diff
@@ -54,7 +60,7 @@ public class OpenTelemetryInstrumentationOptions
     /// </summary>
     public Action<TracerProviderBuilder> BuildTraceCallback { get; set; }
 
-    public void SetMaxBodySize(string val)
+    public static void SetMaxBodySize(string val)
```
Comment (zhenlei520) [inline src/Contrib/StackSdks/Masa.Contrib.StackSdks.Tsc/Tracing/OpenTelemetryInstrumentationOptions.cs]: meaningless parameter name val

File: src/Contrib/StackSdks/Masa.Contrib.StackSdks.Tsc/Tracing/ActivityExtensions.cs (hunk 3 above)
Comment (zhenlei520) [reply src/Contrib/StackSdks/Masa.Contrib.StackSdks.Tsc/Tracing/ActivityExtensions.cs:97]: SetActivityBody(activity, httpRequest.Body, GetHttpRequestEncoding(httpRequest)).Wait();

Synchronous call to asynchronous? Use wait directly? on line19，line 34

**Review by zhenlei520 — CHANGES_REQUESTED** [review]

**Review by zhenlei520 — APPROVED** [review]
LGTM @MayueCif

**Review by MayueCif — APPROVED** [review]
LGTM

**Pull request merged**